### DISCUSSION TITLE
Supporting Pixel Aspect Ratio (PAR) Simulation

### DISCUSSION THREAD

**Discussion by blue-j** [openseadragon/openseadragon#2021] [Ideas]
When people create images for use in video, and/or HDR images, they often have to use pixels that are not square.  This is the related menu in Photoshop:

<img width="526" alt="Screen Shot 2021-08-19 at 4 03 49 PM" src="https://user-images.githubusercontent.com/3990142/130155177-683d57e7-6be9-4eb7-aa30-db61985ec843.png">

You can choose a non-square PAR here, and then turn on "Pixel Aspect Ration Correction" in the same menu to have it simulated.  It's a very cool feature, and I'm wondering about its potential inclusion into OSD.

PAR is detectable only in metadata.  I do not know if OSD has the capacity to extract any metadata?  If not, it could be a switch or flag the user has to insert.  I have done all the work to know where the metadata resides:

# PAR information tags ordered by precedence
'Photoshop:PixelAspectRatio',
'OpenEXR:PixelAspectRatio',
'Radiance:PixelAspectRatio',
'GIF:PixelAspectRatio',
'MakerNotes:PixelAspectRatio'

These group names are from ExifTool, but any metadata library would likely have something similar.

Thanks for considering this idea!  : )

J

**Comment by iangilman** [comment]
Interesting! My intuition is that it would be a good deal of work to pipe this through the drawing system. We don't currently allow for stretching; if we did, it probably would be fairly easy. 

What's the scenario where you would want this? I think in general my suggestion would be to detect the PAR when first processing the image and convert it into a square pixel DZI.

**Reply by blue-j** [reply]
It doesn't / shouldn't involve any editing.  This is only about a simulation in rendering, similar to, say auto-rotation based on EXIF tags.

<google-sheets-html-origin>

Original Width | Original Height | PAR | PAR Corrected Render Width | PAR Corrected Render Height
-- | -- | -- | -- | --
100 | 100 | 0.91 | 91 | 100
100 | 100 | 1.09 | 109 | 100
100 | 100 | 1.33 | 133 | 100
100 | 100 | 1.46 | 146 | 100
100 | 100 | 2 | 200 | 100
100 | 100 | 1.5 | 150 | 100

**Reply by blue-j** [reply]
https://en.wikipedia.org/wiki/Pixel_aspect_ratio

https://helpx.adobe.com/photoshop/using/creating-images-video.html

It is used mostly when people prepare images for use in video - not a miniscule edge case, but not a central use case either.

**Reply by blue-j** [reply]
<img width="1048" alt="Screen Shot 2020-10-17 at 2 44 26 PM" src="https://user-images.githubusercontent.com/3990142/130281376-17b4ae95-6d2e-4ca9-a251-b553d5e5dade.png">
<img width="1077" alt="Screen Shot 2020-10-17 at 2 44 51 PM" src="https://user-images.githubusercontent.com/3990142/130281386-e12edd55-f4c9-40fe-bded-d1c07f015b3c.png">
<img width="1089" alt="Screen Shot 2020-10-17 at 2 45 07 PM" src="https://user-images.githubusercontent.com/3990142/130281387-ce74ec00-0446-4548-bdbe-b3e06f859306.png">
<img width="1059" alt="Screen Shot 2020-10-17 at 2 45 21 PM" src="https://user-images.githubusercontent.com/3990142/130281392-463537c6-629f-441a-9a27-f4ec7e57392d.png">
<img width="896" alt="Screen Shot 2020-10-17 at 2 46 21 PM" src="https://user-images.githubusercontent.com/3990142/130281395-8fbaeb27-375f-4006-b45f-95572fe6f542.png">
<img width="916" alt="Screen Shot 2020-10-17 at 2 46 55 PM" src="https://user-images.githubusercontent.com/3990142/130281396-10dfd810-efb8-4ceb-b4d7-d899a495dac5.png">
<img width="912" alt="Screen Shot 2020-10-17 at 2 47 08 PM" src="https://user-images.githubusercontent.com/3990142/130281400-92022837-e84a-4b52-8db9-e302982bea8f.png">

**Comment by iangilman** [comment]
Thank you for the additional information! OSD is really just for display... it's not anything other than the endpoint of an image pipeline. Often the image is converted specifically for OSD viewing (using a format like DZI or IIIF). For that reason, I think it's not necessary for us to support preview modes like this. We shall see, though!
